.circleci/config.yml

```diff
@@ -52,6 +56,12 @@ jobs:
       - run:
           name: Running tests
           command: poetry run pytest --cov=up42/ --cov-report=xml:.coverage-reports/coverage.xml --durations=3
+      - run:
```

Comment:
@seedlit I guess this is a possible reason why it doesn't end. I'd suggest to separate matrix running for 3.10-12 into a separate step and run it after sonar

Reply:
good guess!